docs/build/use-cases/key-generation.mdx

```diff
@@ -0,0 +1,576 @@
+          "Content-Length": Buffer.byteLength(body).toString()
+        }
+      },
+      (res) => {
+        let data = "";
+        res.setEncoding("utf8");
+        res.on("data", (c) => (data += c));
+        res.on("end", () => {
+          try {
+            const { key } = JSON.parse(data);
+            if (!key) throw new Error(`Bad response: ${data}`);
+            resolve(key.startsWith("0x") ? key : `0x${key}`);
+          } catch (e) {
+            reject(e);
+          }
+        });
+      }
+    );
+    req.on("error", reject);
+    req.write(body);
+    req.end();
+  });
+}
+
+export async function getAppId(): Promise<string> {
+  if (!existsSync(APPD_SOCKET)) {
+    throw new Error("appd socket missing: /run/rofl-appd.sock");
+  }
+  return new Promise((resolve, reject) => {
+    const req = request(
+      { method: "GET", socketPath: APPD_SOCKET, path: "/rofl/v1/app/id" },
+      (res) => {
+        let data = "";
+        res.setEncoding("utf8");
+        res.on("data", (c) => (data += c));
+        res.on("end", () => {
+          const id = data.trim();
+          if (!/^rofl1[0-9a-z]+$/.test(id)) {
+            return reject(new Error(`Bad app id: ${data}`));
+          }
+          resolve(id);
+        });
+      }
+    );
+    req.on("error", reject);
+    req.end();
+  });
+}
+
+export async function getEvmPrivateKey(keyId: string): Promise<string> {
+  try {
+    return await generateKey(keyId, "secp256k1");
+  } catch (e) {
+    const allow = process.env.ALLOW_LOCAL_DEV === "true";
+    const pk = process.env.LOCAL_DEV_PK;
+    if (allow && pk && /^0x[0-9a-fA-F]{64}$/.test(pk)) return pk;
+    throw e;
+  }
+}
+```
+
+</details>
+
+### `src/evm.ts` — ethers helpers
+
+<details>
+  <summary>src/evm.ts</summary>
+
+```ts
+import {
+  JsonRpcProvider,
+  Wallet,
+  parseEther,
+  type TransactionReceipt
+} from "ethers";
+
+export function makeProvider(rpcUrl: string, chainId: number) {
+  return new JsonRpcProvider(rpcUrl, chainId);
+}
+
+export function connectWallet(
+  pkHex: string,
+  rpcUrl: string,
+  chainId: number
+): Wallet {
+  return new Wallet(pkHex).connect(makeProvider(rpcUrl, chainId));
+}
+
+export async function signPersonalMessage(w: Wallet, msg: string) {
+  return w.signMessage(msg);
+}
+
+export async function sendEth(
+  w: Wallet,
+  to: string,
+  amountEth: string
+): Promise<TransactionReceipt> {
+  const tx = await w.sendTransaction({ to, value: parseEther(amountEth) });
+  const rcpt = await tx.wait();
+  if (!rcpt) throw new Error("tx dropped before confirmation");
+  return rcpt;
+}
+```
+
+</details>
+
+### `src/keys.ts` — tiny helpers
+
+<details>
+  <summary>src/keys.ts</summary>
+
+```ts
+import { Wallet, getAddress } from "ethers";
+
+export function privateKeyToWallet(pkHex: string): Wallet {
+  return new Wallet(pkHex);
+}
+
+export function checksumAddress(addr: string): string {
+  return getAddress(addr);
+}
+```
+
+</details>
+
+### `src/server.ts` — minimal HTTP API
```

Comment:
I think running another HTTP API is not a good idea. There is a REST server already running and accessible at `/run/rofl/appd.sock` for this purpose.

Users that want to have EVM-signing capability would need to integrate your HTTP API anyway, so why not connecting to appd directly? Instead of server.ts I'd write a simple test that uses appd, get the key, sign an EVM transaction and submit it to Base.